openmetadata-service/src/main/java/org/openmetadata/service/resources/types/TypeResource.java

```diff
@@ -521,6 +525,89 @@ public Response getAllCustomPropertiesByEntityType(
     }
   }
 
+  @GET
+  @Path("/customProperties/search")
+  @Operation(
+      operationId = "searchCustomProperties",
+      summary = "Search Custom Properties",
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "search response",
+            content =
+                @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = SearchResponse.class)))
+      })
+  public Response searchCustomProperties(
```

Comment:
why does this requires another endpoint and why can't be driven the same search endpoint we have?

Reply:
@harshach  this is because of separation of concern, to create custom properties search , we will need some information from the propertyConfig itself, and it can be heavy processing of query on ui side to avoid that supporting a new api